tests/settings.py

```diff
@@ -0,0 +1,19 @@
+class Settings(object):
```

Comment:
Using a static settings class seems error-prone. How about a `SettingsFactory`? I suspect there's a good reason you did it this way but I think you could:

- Generate `settings` from the `SettingsFactory` based on the instrument info in `main()`.
- Pass the `settings` (perhaps `instrument_settings` to be less generic?) to `run_tests`
- Pass the `settings` along to the various test cases. You've already shown you can pass args to test cases in the `ComponentTests` and `SynopticTests`

It would make things far more traceable than using Settings as a static class.

Already mentioned, but changing public attributes individually when they are causally linked (e.g. `instrument`, `name`, `pv_prefix`) isn't an effective API

Reply:
I remember discussing this at the code review, and I think we decided that the additional boilerplate for passing arguments to each of the test classes was not worth it. The API has been reworked slightly as part of another comment.